### PR TITLE
Integration of M-Elo

### PR DESCRIPTION
**Pull request by Nimeggis**
## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: additionalResolvers/flashcard.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+                context,
+                info});
+
+            let flashcardInput=_args.flashcardInput;
+            for (let item of updatedItems.updateAssessment.items) {
+                if (!oldItems.some(oldItem => oldItem.id === item.id)) {
+                    flashcardInput.itemId= item.id;
+                    break;
+                }
+            }
```
Comment (myluki2000) [inline additionalResolvers/flashcard.ts:230]: Nitpick: This could be written in the following way, which would be a bit more concise & intuitive to understand what's happening, at least to me:

```js
flashcardInput.itemId = updatedItems.updateAssessment.items
.find(item => !oldItems.some(oldItem.id === item.id))
.id;
```

File: additionalResolvers/flashcard.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+                    }
+                }
+            }`;
+            let flashcard = await context.FlashcardService.Mutation.mutateFlashcardSet({
+                root,
+                 args: {
+                    assessmentId:_args.assessmentId
+                 },
+                 // we need to define a selection set manually here, otherwise it thinks we don't need any data
+                 // from this mutation and it won't actually be executed
```
Comment (myluki2000) [inline additionalResolvers/flashcard.ts:256]: This code comment isn't really accurate, is it? You need the selection set to even mutate at all, as it contains the data you want to mutate

File: additionalResolvers/flashcard.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                 selectionSet: selectionSetFlashcard,
+                 context,
+                info});
+            let returnItem;
+            for (let item of updatedItems.updateAssessment.items) {
+                if (!oldItems.some(oldItem => oldItem.id === item.id)) {
+                    returnItem= item;
+                    break;
+                }
+            }
```
Comment (myluki2000) [inline additionalResolvers/flashcard.ts:266]: Could be modified in the same way as above.

File: additionalResolvers/quiz.ts [hunk 4]
```diff
@@ -0,0 +1,1820 @@
...
+                    context,
+                    info});
+
+                let questionInput=_args.questionInput;
+                for (let item of updatedItems.updateAssessment.items) {
+                    if (!oldItems.some(oldItem => oldItem.id === item.id)) {
+                        questionInput.itemId= item.id;
+                        break;
+                    }
+                }
```
Comment (myluki2000) [inline additionalResolvers/quiz.ts:99]: Same as in other comments.

File: additionalResolvers/quiz.ts [hunk 5]
```diff
@@ -0,0 +1,1820 @@
...
+                let selectionSetQuiz=`{_internal_noauth_addMultipleChoiceQuestion(input: {
+                    itemId: "${questionInput.itemId}",
```
Comment (myluki2000) [inline additionalResolvers/quiz.ts:101]: Why does it expect an ID to be passed for the question? The question doesn't exist yet, so there is no ID and shouldn't the `_internal_noauth_addMultipleChoiceQuestion` mutation itself create a new UUID for the question? (This would however make it necessary that the code which updates the skill metadata of this assessment in the content service is called AFTER creating the question. Would that be a problem?) 

Firstly creating the item itself and get its id and then, afterwards, updating the metadata in the content service, would make more sense IMO and would also be consistent with how other parts of the system work, e.g. flashcards and quizzes themselves are first created and then their ID passed to the content service. Why flip this for the assessment items?

**Review by myluki2000 — CHANGES_REQUESTED** [review]
